http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/AKSInterpreter.scala

```diff
@@ -954,14 +960,43 @@ class AKSInterpreter[F[_]](config: AKSInterpreterConfig,
     }
   }
 
+  private[util] def retrieveWsmNamespace(resourceApi: ResourceApi,
+                                         namespacePrefix: String,
+                                         workspaceId: UUID
+  ): F[Option[WsmControlledKubernetesNamespaceResource]] = {
+
+    val wsmNamespaces = F.blocking(
+      resourceApi
+        .enumerateResources(workspaceId, 0, 100, ResourceType.AZURE_KUBERNETES_NAMESPACE, StewardshipType.CONTROLLED)
+        .getResources
+        .asScala
+        .toList
+    )
+    wsmNamespaces.map { namespaces =>
+      // there should be only 1 kubernetes namespace per app
+      namespaces
+        .find(ns =>
```

Comment:
hmm... should we have some sort of exception and error logging here if for some reason we get back more than one namespace that fits this criteria? Seems like a contract against business logic that an external system enforces which would be difficult to diagnose if it changed based on how this is written

Reply:
yeahhh I like that idea, I can add an exception if there's multiple